### PR TITLE
Fix some mirror URLs

### PR DESCRIPTION
**Pull request by mss**
Some source packages went away or were moved around.

### CODE REVIEW

File: thirdparty/libltdl/Makefile [hunk 1]
```diff
@@ -7,7 +7,7 @@ pname := libltdl
 uname := libtool
 pfile := $(uname)-$(pvers).tar.gz
 psrc_file := $(SRC_DIR)/$(pfile)
-purl := http://reflection.oss.ou.edu/gnu/gnu/libtool/$(pfile)
+purl := http://ftpmirror.gnu.org/libtool/$(pfile)
```
Comment (plobbes) [inline thirdparty/libltdl/Makefile]: I know https://www.gnu.org/server/mirror.html says use http://ftpmirror... but it appears they do have/support https so please change the URL here to be https://...

File: thirdparty/libltdl/zimbra-libltdl/debian/watch [hunk 2]
```diff
@@ -1,2 +1,2 @@
 version=3
-http://reflection.oss.ou.edu/gnu/gnu/libtool/libtool-(\d+\.\d+\.\d+).tar.gz
+http://ftpmirror.gnu.org/libtool/libtool-(\d+\.\d+\.\d+).tar.gz
```
Comment (plobbes) [inline thirdparty/libltdl/zimbra-libltdl/debian/watch]: Ditto, please change http: to https:

File: thirdparty/libsodium/Makefile [hunk 3]
```diff
@@ -6,7 +6,7 @@ pvers := $(SODIUM_VERSION)
 pname := libsodium
 pfile := $(pname)-$(pvers).tar.gz
 psrc_file := $(SRC_DIR)/$(pfile)
-purl := https://download.libsodium.org/libsodium/releases/$(pfile)
+purl := https://download.libsodium.org/libsodium/releases/old/unsupported/$(pfile)
```
Comment (plobbes) [inline thirdparty/libsodium/Makefile]: Sigh, this is the wrong "fix"... we need to upgrade to a supported/recent libsodium.

Comment (mss) [reply thirdparty/libsodium/Makefile]: Yes, that's what I noted in the commit message :-) But this allows one to at least do a full-tree rebuild.

I opened [bug 108832](https://bugzilla.zimbra.com/show_bug.cgi?id=108832) regarding the upgrade.

**Comment by mss** [comment]
Any chance to get this merged? The libsodium issue should be handled separately I think.

**Comment by fatkudu** [comment]
Thanks @mss Merged separately.

**Pull request closed without merging**